utils_nlp/models/transformers/abstractive_summarization_bertsum.py

```diff
@@ -0,0 +1,815 @@
+
+    def __init__(
+        self,
+        processor,
+        model_name="bert-base-uncased",
+        finetune_bert=True,
+        cache_dir=".",
+        label_smoothing=0.1,
+        test=False,
+        max_pos_length=768,
+    ):
+        """Initialize an object of BertSumAbs.
+
+        Args:
+            processor (BertSumAbsProcessor): A processor with symbols, tokenizers
+                and collate functions that are used in finetuning and prediction.
+            model_name (str, optional:) Name of the pretrained model which is used
+                to initialize the encoder of the BertSumAbs model.
+                check MODEL_CLASS for supported models. Defaults to "bert-base-uncased".
+            finetune_bert (bool, option): Whether the bert model in the encoder is
+                finetune or not. Defaults to True.
+            cache_dir (str, optional): Directory to cache the tokenizer. Defaults to ".".
+            label_smoothing (float, optional): The amount of label smoothing. Value range is [0, 1]. Defaults to 0.1.
+            test (bool, optional): Whether the class is initiated for test or not.
+                Defaults to False.
+            max_pos_length (int, optional): maximum postional embedding length for the
+                input. Defaults to 768.
+        """
+
+        super().__init__(
+            model_class=MODEL_CLASS,
+            model_name=model_name,
+            num_labels=0,
+            cache_dir=cache_dir,
+        )
+        if model_name not in self.list_supported_models():
+            raise ValueError(
+                "Model name {} is not supported by BertSumAbs. "
+                "Call 'BertSumAbs.list_supported_models()' to get all supported model "
+                "names.".format(value)
+            )
+
+        self.model_class = MODEL_CLASS[model_name]
+        self.cache_dir = cache_dir
+        self.max_pos_length = max_pos_length
+
+        self.model = AbsSummarizer(
+            temp_dir=cache_dir,
+            finetune_bert=finetune_bert,
+            checkpoint=None,
+            label_smoothing=label_smoothing,
+            symbols=processor.symbols,
+            test=test,
+            max_pos=self.max_pos_length,
+        )
+        self.processor = processor
+        self.optim_bert = None
+        self.optim_dec = None
+
+    @staticmethod
+    def list_supported_models():
+        return list(MODEL_CLASS.keys())
+
+    def fit(
+        self,
+        train_dataset,
+        num_gpus=None,
+        gpu_ids=None,
+        batch_size=4,
+        local_rank=-1,
+        max_steps=5e4,
+        warmup_steps_bert=20000,
+        warmup_steps_dec=10000,
+        learning_rate_bert=0.002,
+        learning_rate_dec=0.2,
+        optimization_method="adam",
+        max_grad_norm=0,
+        beta1=0.9,
+        beta2=0.999,
+        decay_method="noam",
+        gradient_accumulation_steps=2,
+        report_every=10,
+        save_every=100,
```

Comment:
yes, it will.  can make it a larger number. How about 1000?

Reply:
Yeah, that would be much better. 
Also not sure if it will help to have an option to save the final model although the users can always call save_model by themselves, up to you.

Reply:
added saving the model at the end of fit function as well

Reply:
Where did you add it? Just be careful and make sure local_rank is -1 or 0. Otherwise, it causes problem when you do distributed training. I've encountered that problem before.

Reply:
check out the newest commit. I don't understand " Just be careful and make sure local_rank is -1 or 0. Otherwise, it causes problem when you do distributed training". Is this comment related to saving models at the end of fit function?